attendance/lib/backend/scan.dart

```diff
@@ -68,6 +74,33 @@ class Scan {
       throw Exception('session already left');
     }
   }
+
+  /// save data to api
+  Future<void> save() async {
+    _persist();
+    try {
+      await _api.addSession(
+        Session.fromMap(<String, dynamic>{
+          'classkey' : classKey,
+          'myKey' : key,
+          'admin' : admin
+        }),
+        await User.token()
+      );
+      _markSaved();
+    } on Exception catch (e) {
+      print(e.toString());
+    }
+  }
+
+  void _persist() {
+    DBProvider.db.newScan(this);
```

Comment:
@ammaratef45 Don't you think it'd be more expressive and if 'newScan' renamed to 'initScan'? 
It'd be less confusing with the original 'new' keyword in readability, i believe!

Reply:
@Bahaaib I agree with you, yet I will edit it and add more unit tests.
Also, The puzzle isn't deleted, I will wait to merge #90 first :)

Reply:
@ammaratef45 OKay